theano/gpuarray/c_code/dnn_fwd.c

```diff
@@ -309,7 +333,52 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
       cuda_exit(c->ctx);
       return 1;
     }
+    // save worksize for next time/cache
+    prev_algo.wsSize = worksize;
 
+    // Add to the cache
+    if (params->choose_algo)
```

Comment:
@nouiz: actually, this is not about 'algo-per-layer' : I don't think we should cache user-specified algos as they may not be optimal. However, if it's guess_once - locally or globally - we do want to cache it so that next time we create the module with same parameters (real life use case) we don't need to run benchmarks.